apiserver/remoterelations/remoterelations.go

```diff
@@ -455,32 +455,26 @@ func (api *RemoteRelationsAPI) registerRemoteRelation(relation params.RegisterRe
 	// TODO(wallyworld) - do this as a transaction so the result is atomic
 	// Perform some initial validation - is the local application alive?
 
-	// The name the consuming side knows the application by is not necessarily
-	// what it has been deployed as locally.
-	localApplicationName := relation.OfferedApplicationName
-	appOffer, err := api.st.ListOffers(crossmodel.ApplicationOfferFilter{
-		ApplicationName: relation.OfferedApplicationName,
+	// Look up the offer record so get the local application to which we need to relate.
+	appOffers, err := api.st.ListOffers(crossmodel.ApplicationOfferFilter{
+		OfferName: relation.OfferName,
 	})
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	if len(appOffer) == 0 {
-		// TODO(wallyworld) - we don't yet record the offer, assume the URL contains the name
-		// return errors.NotFoundf("offered application %q", relation.OfferedApplicationName)
-		appNameParts := strings.Split(relation.OfferedApplicationName, "-")
-		if len(appNameParts) > 1 {
-			localApplicationName = appNameParts[len(appNameParts)-1]
-		}
-	} else {
-		localApplicationName = appOffer[0].ApplicationName
+	if len(appOffers) == 0 {
+		return nil, errors.NotFoundf("application offer %v", relation.OfferName)
 	}
+	localApplicationName := appOffers[0].ApplicationName
 
 	localApp, err := api.st.Application(localApplicationName)
 	if err != nil {
-		return nil, errors.Annotatef(err, "cannot get application for offer %q", relation.OfferedApplicationName)
+		return nil, errors.Annotatef(err, "cannot get application for offer %q", relation.OfferName)
 	}
 	if localApp.Life() != state.Alive {
-		return nil, errors.NotFoundf("application %v", localApplicationName)
+		// We don't want to leak the application name so just log it.
```

Comment:
I didn't understand this at first but I think I do now: this is in API for the offering model, so the local application name is private information - returning it in the error would leak it to the client, but logging it doesn't.